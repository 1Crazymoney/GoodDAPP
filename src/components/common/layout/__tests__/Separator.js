import React from 'react'
import renderer from 'react-test-renderer'
import ImportedSeparator from '../Separator'
import { withThemeProvider } from '../../../../__tests__/__util__'
const Separator = withThemeProvider(ImportedSeparator)

// Note: test renderer must be required after react-native.

describe('Separator', () => {
  it('renders without errors', () => {
    const tree = renderer.create(<Separator />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<Separator />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot with properties', () => {
    const component = renderer.create(<Separator width={3} color="surface" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
