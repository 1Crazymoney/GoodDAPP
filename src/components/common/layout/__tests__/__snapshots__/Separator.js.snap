// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "borderBottomColor": "rgba(0,175,255,1.00)",
        "borderBottomWidth": "1px",
      }
    }
  />
</div>
`;

exports[`Separator matches snapshot with properties 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitBoxFlex": 1,
      "WebkitFlexBasis": "0%",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "flexBasis": "0%",
      "flexGrow": 1,
      "flexShrink": 1,
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msFlexPreferredSize": "0%",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "borderBottomColor": "rgba(255,255,255,1.00)",
        "borderBottomWidth": "3px",
      }
    }
  />
</div>
`;
